mlflow/pyfunc/__init__.py

```diff
@@ -1926,9 +1926,13 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                     return [i.upper() for i in model_input]
 
 
-            mlflow.pyfunc.save_model("model", python_model=MyModel(), input_example=["a"])
-            model = mlflow.pyfunc.load_model("model")
-            print(model.predict(["a", "b", "c"]))  # -> ["A", "B", "C"]
+            my_model = MyModel()
+            with mlflow.start_run():
+                model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model)
```

Comment:
```suggestion
            with mlflow.start_run():
                model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=MyModel(), input_example=["a"])
```
This extra variable definition seems useless, and we can still keep input_example :)

Reply:
There are two remarks here, the extra variable definition is indeed useless. But I stole it from [another documentation section](https://github.com/mlflow/mlflow/blob/19079d163a7569b759759e6cc2518bb18f62ce1d/mlflow/pyfunc/__init__.py#L498).

using the `input_example`-arg raises an error. So this needs to be fixed too, that is why I left it out.

```
Traceback (most recent call last):
  File "/home/bramdj/anaconda3/envs/mlflow-dev-env/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/bramdj/anaconda3/envs/mlflow-dev-env/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/home/bramdj/.vscode/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/home/bramdj/Desktop/temp$.py", line 12, in <module>
    model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model, input_example=["a"])
  File "/home/bramdj/Desktop/git/mlflow/mlflow/pyfunc/__init__.py", line 2024, in log_model
    return Model.log(
  File "/home/bramdj/Desktop/git/mlflow/mlflow/models/model.py", line 579, in log
    flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
  File "/home/bramdj/Desktop/git/mlflow/mlflow/pyfunc/__init__.py", line 1819, in save_model
    if signature := _infer_signature_from_type_hints(
  File "/home/bramdj/Desktop/git/mlflow/mlflow/models/signature.py", line 310, in _infer_signature_from_type_hints
    output_example = func(input_example) if input_example else None
TypeError: predict() missing 1 required positional argument: 'model_input'
``

Reply:
I've made some changes to remove the extra variable definition, if you want I can add `input_example` too but that is up to you

Reply:
DCO check passes to, I was apparently not consistent in my sign offs. My apologies

Reply:
Oh this sounds like a problem... We should definitely fix it.
In that case, feel free to remove the input_example in this docs, we can add it back later if needed. Thanks!

Reply:
If I have time to dig up some more information, I'll open a new issue for that.

Reply:
@serena-ruan, I have signed off on everything and made the requested changes, feel free to merge when all checks pass.

Reply:
Thank you!